### PR TITLE
feat: [PIE-3737]: Add lock step to step library

### PR DESCRIPTION
**Pull request by Fernando-Dourado**
Please read [go/trigger-cmd](https://harness.atlassian.net/wiki/spaces/BT/pages/21082441647/PR+Checks) to re-trigger PR Checks using Github comments.

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/33516)

### CODE REVIEW

File: 860-orchestration-steps/src/main/java/io/harness/plancreator/steps/resourceconstraint/LockStepInfo.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+   *     <li>name: {@link PmsConstants#QUEUING_RC_NAME}</li>
+   *     <li>permits: {@link PmsConstants#QUEUING_RC_PERMITS}</li>
+   *     <li>acquireMode: {@link AcquireMode#ENSURE} (FIFO)</li>
+   * </ul>
+   */
+  @Override
+  public SpecParameters getSpecParameters() {
+    return ResourceRestraintSpecParameters.builder()
+        .resourceUnit(key.getValue())
+        .holdingScope(HoldingScope.builder().scope("PLAN").build()) // COME FROM UI
```
Comment (naidu-harness) [inline 860-orchestration-steps/src/main/java/io/harness/plancreator/steps/resourceconstraint/LockStepInfo.java:69]: I think this scope is changing to PIPELINE/STAGE.
@vaibhav-si @hextechpal when are we planning to merge the PR?

Comment (vaibhav-si) [reply 860-orchestration-steps/src/main/java/io/harness/plancreator/steps/resourceconstraint/LockStepInfo.java:69]: We need to send scope from step Info

Comment (vaibhav-si) [reply 860-orchestration-steps/src/main/java/io/harness/plancreator/steps/resourceconstraint/LockStepInfo.java:69]: Also, we can't do `getValue()` as expressions will not be resolved at this point. We need to send parameterfield as it is to specParams.

Comment (Fernando-Dourado) [reply 860-orchestration-steps/src/main/java/io/harness/plancreator/steps/resourceconstraint/LockStepInfo.java:69]: We already discussed that, for now, it's here to unblock UI team until I merge the most recent changes from PP where HoldingScope changes to an enum.

**Comment by Fernando-Dourado** [comment]
trigger ti1

**Comment by Fernando-Dourado** [comment]
A new PR is open to easily review and merge.

**Pull request closed without merging**